tabular/src/autogluon/tabular/predictor/predictor.py

```diff
@@ -740,11 +755,40 @@ def fit(
+            ds_args : dict, see bleow for default
+                Keyword arguments for dynamic stacking, only used if `dynamic_stacking=True`. These keyword arguments control the behavior of dynamic stacking
+                and determine how AutoGluon tries to detect stacked overfitting. To detect stacked overfitting, AutoGluon will fit itself (so called sub-fits)
+                on a subset (for holdout) or multiple subsets (for repeated cross-validation) and use the predictions of AutoGluon on the validation data to
+                detect stacked overfitting. The sub-fits stop and stacking will be disabled if any sub-fit shows stacked overfitting.
+                Allowed keys and values are:
+                    `detection_time_frac` : float in (0,1), default = 1/4
+                        Determines how much of the original training time is used for detecting stacked overfitting.
+                        If no time limit is given to AutoGluon, this parameter is ignored and AutoGluon is fit without a time limit in the sub-fit.
+                        When using (repeatd) cross-validation, each sub-fit will be fit for `1/n_splits * detection_time_frac * time_limit`.
+                    `use_holdout` : bool, default = True
+                        Whether to use holdout validation or not.
+                    `holdout_frac` : float in (0,1), default = 1/9
+                        Determines how much of the original training data is used for the holdout data.
+                    `n_folds` : int in [2, +inf), default = 2
+                        Number of folds to use for cross-validation.
+                    `n_repeats` : int [1, +inf), default = 1
+                        Number of repeats to use for repeated cross-validation.
+                    `memory_safe_fits` : bool, default = True
+                        If True, AutoGluon runs each sub-fit in a ray-based subprocess to avoid memory leakage that exist due to Python's lackluster
+                        garbage collector.
+                    `clean_up_fits` : bool, default = True
+                        If True, AutoGluon will remove all saved information from sub-fits from disc.
+                        If False, the sub-fits are kept on disc and `self._sub_fits` will store paths to the sub-fits, which can be loaded just like any other
+                        predictor from disc using `TabularPredictor.load()`.
+                    `custom_val_data`: str or :class:`TabularDataset` or :class:`pd.DataFrame`, default = None
+                        Another dataset containing validation data reserved for detecting stacked overfitting. This dataset should be in the same format as
+                        `train_data`. If str is passed, `custom_val_data` will be loaded using the str value as the file path.
+                        If `custom_val_data` is not None, the sub-fit fit on all of `train_data`.
```

Comment:
Clarify that `custom_val_data` is incorporated into the final fit.
Rename to `tuning_data` to be consistent with the fit() API.

Reply:
`custom_val_data` is so far not incorporated into the final fit. Will add that. But I am not sure about naming `tuning_data`. Tuning data sounds very misleading here.